VideoLocker/src/main/java/org/edx/mobile/util/EmailUtil.java

```diff
@@ -11,32 +11,26 @@
 
     private static final Logger logger = new Logger(EmailUtil.class.getName());
 
-    public static void sendEmail(final FragmentActivity activityContext, final String to,
-                                 final String subject, final String email, Config config) {
+    public static void openEmailClient(final FragmentActivity activityContext, final String to,
+                                       final String subject, final String email, Config config) {
         // verify if the app is running on zero-rated mobile data?
         if (NetworkUtil.isConnectedMobile(activityContext) && NetworkUtil.isOnZeroRatedNetwork(activityContext, config)) {
             // inform user they may get charged for sending email
             IDialogCallback callback = new IDialogCallback() {
                 @Override
                 public void onPositiveClicked() {
-                    openEmailClient(activityContext, to, subject, email);
                 }
 
                 @Override
                 public void onNegativeClicked() {
+                    return;
                 }
             };
-
             MediaConsentUtils.showLeavingAppDataDialog(activityContext, callback);
-        } else {
```

Comment:
The else clause was needed; now were going to both show the consent dialog AND open the email client